core/vcpu.c

```diff
@@ -4311,6 +4316,66 @@ static void vcpu_state_dump(struct vcpu_t *vcpu)
             vcpu->state->_idt.base,
             vcpu->state->_idt.limit,
             (uint32_t)vcpu->state->_efer);
+
+    // Dump FPU state
+    hax_log(HAX_LOGW,
+            "FX layout:\n"
+            "fcw:    %08x  fsw:    %08x  ftw: %08x  res1: %08x  fop: %08x\n"
+            "fpu_ip: %08llx  fpu_dp: %08llx\n"
+            "mxcsr:  %08x  mxcsr_mask: %08x\n",
+            gfx->fcw, gfx->fsw, gfx->ftw, gfx->res1, gfx->fop, gfx->fpu_ip,
+            gfx->fpu_dp, gfx->mxcsr, gfx->mxcsr_mask);
+
+    for (i = 0; i < 16; i++) {
+        hax_log(HAX_LOGW, "st_mm[%d] %02x%02x%02x%02x%02x%02x%02x%02x\n", i,
```

Comment:
Do you mind to add the colon after the array name and add a space every 4 bytes for separation?

Reply:
Like this:
st_mm [%d] %02x%02x%02x%02x %02x%02x%02x%02x\n
?

Reply:
Correct, 4-byte separation is just for reading convenience.